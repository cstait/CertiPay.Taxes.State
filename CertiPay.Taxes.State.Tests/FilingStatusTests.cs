//using CertiPay.Common.Testing;
using CertiPay.Payroll.Common;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Runtime.CompilerServices;

namespace CertiPay.Taxes.State.Tests
{
    public class FilingStatusTests
    {
        [Test, MethodImpl(MethodImplOptions.NoInlining)]
        public void Generate_Status_List()
        {
            var list = new Dictionary<StateOrProvince, IEnumerable<String>>();

            foreach (var state in States.Values())
            {
                if (state.HasWithholding())
                {
                    list.Add(state, FilingStatuses.ForState(state));
                }
            }

            //list.VerifyMe();
        }
    }
}