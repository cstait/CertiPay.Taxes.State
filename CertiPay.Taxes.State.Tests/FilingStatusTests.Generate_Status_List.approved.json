{
    
  "Unknown":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "AK":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "AS":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "AZ":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "AR":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "CA":  [
        
    "Single",
        
    "Married",
        
    "HeadOfHousehold",
        
    "Exempt"
  
    ],
    
  "CO":  [
        
    "Single",
        
    "Married",
        
    "Exempt"
  
    ],
    
  "CT":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "DE":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "DC":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "FM":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "GA":  [
        
    "Single",
        
    "MarriedWithOneIncome",
        
    "MarriedWithTwoIncomes",
        
    "MarriedFilingSeparate",
        
    "HeadOfHousehold",
        
    "Exempt"
  
    ],
    
  "GU":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "HI":  [
        
    "Single",
        
    "Married",
        
    "Exempt"
  
    ],
    
  "ID":  [
        
    "Single",
        
    "Married",
        
    "Exempt"
  
    ],
    
  "IL":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "IN":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "IA":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "KS":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "KY":  [
        
    "Normal",
        
    "% of Gross",
        
    "Exempt"
  
    ],
    
  "LA":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "ME":  [
        
    "Single",
        
    "Married",
        
    "Exempt"
  
    ],
    
  "MH":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "MD":  [
        
    "Single",
        
    "Married",
        
    "Exempt"
  
    ],
    
  "MA":  [
        
    "Single",
        
    "Married",
        
    "HeadOfHousehold",
        
    "Exempt"
  
    ],
    
  "MI":  [
        
    "Normal",
        
    "% of Gross",
        
    "Exempt"
  
    ],
    
  "MN":  [
        
    "Single",
        
    "Married",
        
    "Exempt"
  
    ],
    
  "MS":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "MO":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "MT":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "NE":  [
        
    "Single",
        
    "Married",
        
    "Exempt"
  
    ],
    
  "NJ":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "NM":  [
        
    "Single",
        
    "Married",
        
    "Exempt"
  
    ],
    
  "NY":  [
        
    "Single",
        
    "Married",
        
    "Exempt"
  
    ],
    
  "NC":  [
        
    "Single",
        
    "Married",
        
    "HeadOfHousehold",
        
    "Exempt"
  
    ],
    
  "ND":  [
        
    "Single",
        
    "Married",
        
    "Exempt"
  
    ],
    
  "MP":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "OH":  [
        
    "Normal",
        
    "% of Gross",
        
    "Exempt"
  
    ],
    
  "OK":  [
        
    "Single",
        
    "Married",
        
    "Exempt"
  
    ],
    
  "OR":  [
        
    "Single",
        
    "Married",
        
    "Exempt"
  
    ],
    
  "PW":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "PA":  [
        
    "Normal",
        
    "% of Gross",
        
    "Exempt"
  
    ],
    
  "PR":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "RI":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "SC":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "UT":  [
        
    "Single",
        
    "Married",
        
    "Exempt"
  
    ],
    
  "VT":  [
        
    "Single",
        
    "Married",
        
    "Exempt"
  
    ],
    
  "VI":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "VA":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "WA":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "WV":  [
        
    "Normal",
        
    "Exempt"
  
    ],
    
  "WI":  [
        
    "Single",
        
    "Married",
        
    "Exempt"
  
    ]

}